lib/kotlin/cross-test-client/build.gradle.kts

```diff
@@ -31,6 +31,7 @@ repositories {
 dependencies {
     implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
     implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+    implementation("com.github.ajalt.clikt:clikt:3.4.2")
```

Comment:
What does clikt do?
Is this only a test dependency?

Reply:
@ctubbsii testImplementation is for unit tests, here we are building a test binary (application) for cross test.

In java it's achieved by configuring a shadowJar with customized source set `crossTest` but here we use a different approach (i.e. multi module gradle project). The whole project in Kotlin library is _not_ published, because at runtime it only uses Java libthrift. Here the content under `lib/kotlin` is only for driving cross tests.

Reply:
https://ajalt.github.io/clikt/ clikt is a framework for building CLI apps, it helps with command line parsing and validation.

Reply:
@Jimexist Thanks for the updates to the README. That helps. I should have paid closer attention to that before.